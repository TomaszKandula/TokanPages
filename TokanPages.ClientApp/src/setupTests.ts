// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
