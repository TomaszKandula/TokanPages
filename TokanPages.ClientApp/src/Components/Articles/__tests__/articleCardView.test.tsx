import "../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import ArticleCardView from "../articleCardView";

describe("Test account group component: articleCardView.", () => 
{
    it("Renders correctly '<ArticleCardView />' when content is loaded.", () => 
    {
        const tree = shallow(<ArticleCardView bind=
        {{
            imageUrl: "/images/card.jpg",
            title: "Article title",
            description: "Article short description",
            onClickEvent: jest.fn(),
            buttonText: "READ"
        }} />);
        expect(tree).toMatchSnapshot();
    });
});
