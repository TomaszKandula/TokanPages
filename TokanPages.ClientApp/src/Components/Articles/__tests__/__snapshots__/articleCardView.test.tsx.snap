// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test articles group component: articleCardView. Renders correctly '<ArticleCardView />' when content is loaded. 1`] = `
<div
  data-aos="fade-up"
>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-root-1"
    elevation={4}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={4}
      >
        <img
          alt=""
          className="makeStyles-img-4"
          src="/images/card.jpg"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={8}
      >
        <WithStyles(ForwardRef(CardContent))>
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-title-5"
            component="h2"
            gutterBottom={true}
            variant="h5"
          >
            Article title
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-description-6"
            color="textSecondary"
            component="p"
            variant="body2"
          >
            Article short description
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(CardActions))>
          <WithStyles(ForwardRef(Button))
            color="primary"
            onClick={[MockFunction]}
            size="small"
          >
            READ
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(CardActions))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Card))>
</div>
`;
